### PR TITLE
add a dialect for sparksql

### PR DESCRIPTION
**Pull request by Jeffrio**
recently,we use spark sql
there is no SparkSqlDialect supplied by mondiran
I bulid the spark sql accroding to the hiveDialect, and this work for me , now I get the the spark sql running on saiku with mondrian3.9

### CODE REVIEW

File: src/main/mondrian/olap/Mondrian_SW.xml [hunk 1]
```diff
@@ -1418,6 +1418,7 @@
             <Value>greenplum</Value>
             <Value>vectorwise</Value>
             <Value>hive</Value>
+            <Value>sparksql</Value>
```
Comment (lucboudreau) [inline src/main/mondrian/olap/Mondrian_SW.xml]: You should also update Mondrian.xml. This one here is for the schema workbench.

File: src/main/mondrian/spi/impl/SparkSqlDialect.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+     *
+     * @param connection Connection
+     *
+     * @throws java.sql.SQLException on error
+     */
+    public SparkSqlDialect(Connection connection) throws SQLException {
+        super(connection);
+    }
+
+    protected String deduceIdentifierQuoteString(
```
Comment (lucboudreau) [inline src/main/mondrian/spi/impl/SparkSqlDialect.java]: No quotes at all? This won't work with drillthrough operations, where the column alias might be two words. Not quoting them will create invalid SQL.

Comment (Jeffrio) [inline src/main/mondrian/spi/impl/SparkSqlDialect.java]: spark sql use back-tick as quote

File: src/main/mondrian/olap/Mondrian_SW.xml (hunk 1 above)
Comment (Jeffrio) [inline src/main/mondrian/olap/Mondrian_SW.xml]: I have updated  Mondrian.xml add sparksql

**Comment by lucboudreau** [comment]
Thank you for this contribution. I've created this ticket to track it and our PMs will prioritize this feature.

http://jira.pentaho.com/browse/MONDRIAN-2328

**Comment by SPECUSA** [comment]
Hello @Jeffrio 

I'm trying to add dialect for sparksql in mondrian. I performed the steps as you have shown to add dialect. But it work well in Pentaho CE though it doesn't work in Schema Workbench

I'm using Pentaho CE -5.0.1, Schema Workbench  3.9.0.0-213 and made changes in Mondrian 3.9

Please let me know why I can't get Sparksql dialect on schema workbench

Thanks in advance

**Pull request closed without merging**